pkg/controller/assignmetadata/assignmetadata_controller.go

```diff
@@ -123,6 +123,7 @@ func (r *AssignMetadataReconciler) Reconcile(request reconcile.Request) (reconci
 	mutator, err := mutation.MutatorForAssignMetadata(assignMetadata)
 	if err != nil {
 		log.Error(err, "Creating mutator for resource failed", "resource", request.NamespacedName)
+		return ctrl.Result{}, nil // Do not reque on resource compilation errors
```

Comment:
Can we state with certainty that these errors are never recoverable?

Reply:
The possible errors are:
- failure to get resource metadata
- invalid location
- location invalid for assignmetadata (not withing metadata, ...)

All prevent a mutator from being created.
If continue we'll attempt to Upsert a nil mutator.

We could think about adding some status that would notify about these failure (preferably another issue and PR)

Reply:
+1, we should create an issue to add mutator status fields

Reply:
Opened #997 to start a discussion.
Can you please add the mutation label?

Reply:
Added